index.d.ts

```diff
@@ -0,0 +1,71 @@
+import { EventEmitter } from 'events'
+import { Duplex } from 'stream'
+
+export interface MetamaskInpageProviderOptions {
+    /**
+     * The logging API to use.
+     * @default console
+     */
+    logger?: Pick<Console, 'log' | 'warn' | 'error' | 'debug' | 'info' | 'trace'>
+    /**
+     * The maximum number of event listeners.
+     * @default 100
+     */
+    maxEventListeners?: number
+    /**
+     * Whether the provider should send page metadata.
+     * @default true
+     */
+    shouldSendMetadata?: boolean
+}
+export class MetamaskInpageProvider extends EventEmitter {
+    /**
+     *
+     * @param connectionStream A Node.js duplex stream
+     * @param options An options bag
+     */
+    constructor(connectionStream: Duplex, options?: MetamaskInpageProviderOptions)
+    readonly isMetaMask: true
+    readonly selectedAddress: string | null
+    readonly networkVersion: string | null
+    readonly chainId: string | undefined
+    isConnected(): boolean
+    sendAsync(payload: JsonRpcPayload, callback: (error: Error | null, result?: JsonRpcResponse) => void): void
+    /** Submits an RPC request to MetaMask for the given method, with the given params. Resolves with the result of the method call, or rejects on error. */
+    request(args: JsonRpcPayload): Promise<any>
```

Comment:
We should prefer `unknown`:
```suggestion
    request(args: JsonRpcPayload): Promise<unknown>
```